src/main/java/com/faforever/client/mod/ModManagerController.java

```diff
@@ -86,6 +93,38 @@ public void initialize() {
     loadActivatedMods();
 
     modListView.scrollTo(modListView.getSelectionModel().getSelectedItem());
+
+    modSearchTextField.textProperty().addListener(new ChangeListener<String>() {
+      @Override
+      public void changed(ObservableValue<? extends String> observable, String oldValue, String newValue) {
+        if (newValue.isEmpty()) {
+          modVersionFilteredList.setPredicate(null);
+        } else {
+          modVersionFilteredList.setPredicate(modVersion -> modVersion.getDisplayName().toLowerCase().contains(newValue.toLowerCase()));
+        }
+        if (!modVersionFilteredList.isEmpty()) {
+          modListView.getSelectionModel().select(0);
+        }
+      }
+    });
+    modSearchTextField.setOnKeyPressed(event -> {
+      MultipleSelectionModel<ModVersion> selectionModel = modListView.getSelectionModel();
+      int currentMapIndex = selectionModel.getSelectedIndex();
+      int newMapIndex = currentMapIndex;
```

Comment:
rather mod index?

Reply:
ah right, I think we should refactor this logic out cause this might be repeated elsewhere, but don't know how much work does it require

Reply:
you could extract the event listener into a separate class, passing all arguments into the constructor